include/miral/miral/wayland_extensions.h

```diff
@@ -92,9 +92,24 @@ class WaylandExtensions
         std::function<std::shared_ptr<void>(Context const* context)> build;
     };
 
+    /// Information that can be used to determine if to enable a conditionally enabled extension
+    /// \remark Since MirAL 3.4
+    struct Info
+    {
+        /// The application that is being given access to this extension
+        Application const& app;
+        /// If this extension is in the recommended set
+        bool recommended;
+        /// If the user has enabled this extension using the wayland-extensions or related option
+        bool manually_enabled;
+    };
+
```

Comment:
Structs with data members can become problematic for maintaining ABI. When there are changes the code creating or copying the `struct` need not have been built with the same definition as that using it.

Reply:
Done